core/templates/pages/learner-group-pages/create-group/add-syllabus-items.component.ts

```diff
@@ -0,0 +1,383 @@
+  activeMenuName: string = '';
+  searchIsInProgress = false;
+  syllabusFilter!: LearnerGroupSyllabusFilter;
+  debounceTimeout = 1000;
+
+  constructor(
+    private windowDimensionsService: WindowDimensionsService,
+    private translateService: TranslateService,
+    private navigationService: NavigationService,
+    private languageUtilService: LanguageUtilService,
+    private constructTranslationIdsService: ConstructTranslationIdsService,
+    private learnerGroupSyllabusBackendApiService:
+      LearnerGroupSyllabusBackendApiService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  isMobileViewActive(): boolean {
+    return this.windowDimensionsService.getWidth() <= 766;
+  }
+
+  ngOnInit(): void {
+    this.SEARCH_DROPDOWN_CATEGORIES = this.searchDropdownCategories();
+    this.KEYBOARD_EVENT_TO_KEY_CODES = (
+      this.navigationService.KEYBOARD_EVENT_TO_KEY_CODES);
+    this.ACTION_OPEN = this.navigationService.ACTION_OPEN;
+    this.ACTION_CLOSE = this.navigationService.ACTION_CLOSE;
+    this.SUPPORTED_CONTENT_LANGUAGES = (
+      this.languageUtilService.getLanguageIdsAndTexts());
+    this.SEARCH_DROPDOWN_TYPES = this.searchDropdownTypes();
+    this.selectionDetails = {
+      types: {
+        description: '',
+        itemsName: 'types',
+        masterList: this.SEARCH_DROPDOWN_TYPES,
+        selection: '',
+        defaultValue: 'All',
+        summary: 'Type'
+      },
+      categories: {
+        description: '',
+        itemsName: 'categories',
+        masterList: this.SEARCH_DROPDOWN_CATEGORIES,
+        selection: '',
+        defaultValue: 'All',
+        summary: 'Category'
+      },
+      languageCodes: {
+        description: '',
+        itemsName: 'languages',
+        masterList: this.SUPPORTED_CONTENT_LANGUAGES,
+        selection: '',
+        defaultValue: 'All',
+        summary: 'Language'
+      }
+    };
+    // Initialize the selection descriptions and summaries.
+    for (let itemsType in this.selectionDetails) {
+      this.updateSelectionDetails(itemsType);
+    }
+
+    this.refreshSearchBarLabels();
+
+    this.searchQueryChanged
+      .pipe(debounceTime(this.debounceTimeout))
+      .pipe(distinctUntilChanged())
+      .subscribe(value => {
+        this.searchQuery = value;
+        this.onSearchQueryChangeExec();
+      });
+
+    this.directiveSubscriptions.add(
+      this.translateService.onLangChange
+        .subscribe(() => this.refreshSearchBarLabels()));
+  }
+
+  // Update the description field of the relevant entry of selectionDetails.
+  updateSelectionDetails(itemsType: string): void {
+    let masterList = this.selectionDetails[itemsType].masterList;
+
+    let selectedItemText = '';
+    for (let i = 0; i < masterList.length; i++) {
+      if (this.selectionDetails[itemsType].selection === masterList[i].id) {
+        selectedItemText = masterList[i].text;
+      }
+    }
+
+    // TODO(milit): When the language changes, the translations won't
```

Comment:
What is this about?

Reply:
Unwanted comment, removed it as per our discussion. Thanks!